src/test/java/seedu/momentum/model/ModelManagerTest.java

```diff
@@ -94,6 +103,176 @@ public void getFilteredProjectList_modifyList_throwsUnsupportedOperationExceptio
         assertThrows(UnsupportedOperationException.class, () -> modelManager.getDisplayList().remove(0));
     }
 
+    //=========== Undo/Redo ================================================================================
+
+    @Test
+    public void canUndoCommand_cannotUndoCommand_returnsFalse() {
+        // modelManager has just been initialiaed, no history log for undo.
+        assertEquals(modelManager.canUndoCommand(), false);
+    }
+
+    @Test
+    public void canUndoCommand_canUndoCommand_returnsTrue() {
+        modelManager.commitToHistory();
+        assertEquals(modelManager.canUndoCommand(), true);
+    }
+
+    @Test
+    public void canRedoCommand_cannotRedoCommand_returnsFalse() {
+        // modelManager has just been initialized, no history log for redo.
+        assertEquals(modelManager.canRedoCommand(), false);
+    }
+
+    @Test
+    public void canRedoCommand_canRedoCommand_returnsTrue() {
+        modelManager.commitToHistory();
+        modelManager.undoCommand();
+        assertEquals(modelManager.canRedoCommand(), true);
+    }
+
+    @Test
```

Comment:
Does this method need to be annotated with `@Test`? It seems like it's only being called by other test methods? Same for `redoCommand`

Reply:
I might be misunderstanding this though. If it is actually a test, maybe the name can be changed? You have it in `VersionedProjectBookTest` as well

Reply:
Yup they are tests, they check whether the fields of modelmanager have been updated correctly after undo/redo

Reply:
It'll be ideal if they can be named in this manner `featureUnderTest_testScenario_expectedBehavior()`

I think the updateOrder method can follow this convention

Reply:
It is because these methods are only called under 1 scenario (which is that it will work) which is why i didn't put test scenario. I added expected behaviour to the names so they sound more like tests!